### ISSUE TITLE
Mutli-Monitor setup doesn't work

### ISSUE THREAD

**Issue by dmgudkov** [Codectory/HDR-Profile#1]
### Exception
1/15/2021 12:23:15 PM
	Unable to load DLL 'HDRController.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

   at HDRProfile.HDRController.SetHDR(Boolean enabled)
   at HDRProfile.HDRController.ActivateHDR()
   at HDRProfile.HDRProfileHandler.StartApplication(ApplicationItem application)
### 

1.2 and 1.21 release versions

**Comment by Codectory** [comment]
Are you using a 64Bit system with an Nvidia Card?

dmgudkov <notifications@github.com> schrieb am Fr. 15. Jan. 2021 um 18:26:

**Comment by dmgudkov** [comment]
Yes. WIn10 x64 and 1080gtx

**Comment by dmgudkov** [comment]
Actually none of them work. Checked other versions. All crash on an attempt to activate HDR. Probably for the same reason (most of them don't have logs).

**Comment by Codectory** [comment]
I will look into this later.

dmgudkov <notifications@github.com> schrieb am Fr. 15. Jan. 2021 um 18:33:

**Comment by Codectory** [comment]
Please check the newest release.

**Comment by dmgudkov** [comment]
### Exception
1/16/2021 9:54:46 AM
	Unable to load DLL 'HDRController.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

   at HDRProfile.HDRController.SetHDRState(Boolean enabled)
   at HDRProfile.HDRController.ActivateHDR()
   at HDRProfile.HDRProfileHandler.StartApplication(ApplicationItem application)
###

**Comment by Codectory** [comment]
Could you please test toggling HDR on the tray icon and send me the full log file? The HDRController.dll is integrated in the exe file and cannot be missing. Thats a bit strange.

**Comment by dmgudkov** [comment]
[HDRProfile.log](https://github.com/Codectory/HDR-Profile/files/5824584/HDRProfile.log)

Sure thing. Closed to tray. Clicked "Activate HDR" from tray icon context menu.

**Comment by Codectory** [comment]
Let's try to copy the DLL in the same folder as the exe file and run it again.
[HDRController.zip](https://github.com/Codectory/HDR-Profile/files/5824593/HDRController.zip)

**Comment by dmgudkov** [comment]
Already tried before. Tried again. No dice.

You're right, there's something weird here.

Btw, activating from tray for the most part doesn't even output that error.

It just crashes with this:

DESKTOP-O8K6993	HDRPProfile 1.0.0.0

#######################
1/16/2021 10:06:34 AM:Initializing...
1/16/2021 10:06:34 AM:Loading settings..
1/16/2021 10:06:34 AM:Settings loaded
1/16/2021 10:06:34 AM:Initializing tray menu
1/16/2021 10:06:34 AM:Tray menu initialized
1/16/2021 10:06:34 AM:Initialized
1/16/2021 10:06:37 AM:Minimizing to tray...

So the easiest thing how i reproduce the error is to use "Start in HDR mode" button. Then it's 100% reproduceable missing dll error in the logs.

**Comment by dmgudkov** [comment]
Yep. Tried to test more with tray "Activate HDR" and "Deactivate HDR" options. It just crashes without that message in logs.

Not exactly sure how i managed to produce the error in one of the attempts. Might have been a leftover in the logs from starting game/clicking "Start in HDR mode".

**Comment by Codectory** [comment]
I uploaded a 32bit and 64bit version. Could you try both?

**Comment by dmgudkov** [comment]
32-bit:

DESKTOP-O8K6993	HDRPProfile 1.3.2.0

#######################
1/16/2021 10:25:30 AM:Initializing...
1/16/2021 10:25:30 AM:Loading settings..
1/16/2021 10:25:30 AM:Failed to load settings
### Exception
1/16/2021 10:25:30 AM
	Could not find file 'F:\MainDownloads\Release_HDRProfile_1-3-2_x86\HDRProfile_Settings.xml'.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path)
   at HDRProfile.HDRProfileSettings.ReadSettings(String path)
   at HDRProfile.HDRProfileHandler.LoadSettings()
### 
1/16/2021 10:25:30 AM:Saving settings..
1/16/2021 10:25:30 AM:Settings saved
1/16/2021 10:25:30 AM:Created new settings file
1/16/2021 10:25:30 AM:Settings loaded
1/16/2021 10:25:30 AM:Initializing tray menu
1/16/2021 10:25:30 AM:Tray menu initialized
1/16/2021 10:25:30 AM:Initialized
1/16/2021 10:25:31 AM:Starting process watcher...
1/16/2021 10:25:31 AM:Process watcher started
1/16/2021 10:25:32 AM:Saving settings..
1/16/2021 10:25:32 AM:Settings saved
1/16/2021 10:25:33 AM:Saving settings..
1/16/2021 10:25:33 AM:Settings saved
1/16/2021 10:25:33 AM:Saving settings..
1/16/2021 10:25:33 AM:Settings saved
1/16/2021 10:25:35 AM:Updating HDR mode to Running...
### Exception
1/16/2021 10:25:35 AM
	Unable to load DLL 'HDRController.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

   at HDRProfile.HDRController.SetHDRState(Boolean enabled)
   at HDRProfile.HDRController.DeactivateHDR()
   at HDRProfile.HDRProfileHandler.UpdateHDRMode()
### 
1/16/2021 10:25:39 AM:Minimizing to tray...




64-bit:

DESKTOP-O8K6993	HDRPProfile 1.3.2.0

#######################
1/16/2021 10:26:22 AM:Initializing...
1/16/2021 10:26:22 AM:Loading settings..
1/16/2021 10:26:22 AM:Failed to load settings
### Exception
1/16/2021 10:26:22 AM
	Could not find file 'F:\MainDownloads\Release_HDRProfile_1-3-2_x64\HDRProfile_Settings.xml'.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
   at System.IO.StreamReader..ctor(String path)
   at HDRProfile.HDRProfileSettings.ReadSettings(String path)
   at HDRProfile.HDRProfileHandler.LoadSettings()
### 
1/16/2021 10:26:22 AM:Saving settings..
1/16/2021 10:26:22 AM:Settings saved
1/16/2021 10:26:22 AM:Created new settings file
1/16/2021 10:26:22 AM:Settings loaded
1/16/2021 10:26:22 AM:Initializing tray menu
1/16/2021 10:26:23 AM:Tray menu initialized
1/16/2021 10:26:23 AM:Initialized
1/16/2021 10:26:23 AM:Starting process watcher...
1/16/2021 10:26:23 AM:Process watcher started
1/16/2021 10:26:24 AM:Saving settings..
1/16/2021 10:26:24 AM:Settings saved
1/16/2021 10:26:24 AM:Saving settings..
1/16/2021 10:26:24 AM:Settings saved
1/16/2021 10:26:25 AM:Saving settings..
1/16/2021 10:26:25 AM:Settings saved
1/16/2021 10:26:26 AM:Updating HDR mode to Running...
### Exception
1/16/2021 10:26:26 AM
	Unable to load DLL 'HDRController.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

   at HDRProfile.HDRController.SetHDRState(Boolean enabled)
   at HDRProfile.HDRController.DeactivateHDR()
   at HDRProfile.HDRProfileHandler.UpdateHDRMode()
### 
1/16/2021 10:26:28 AM:Minimizing to tray...


In both cases i clicked "Activate HDR" from tray icon menu.

**Comment by dmgudkov** [comment]
Ok.

I did some research.

According Process Monitor it indeed can't find HDRControl.dll

If i put the dll in the folder then it finds it (in Process Monitor logs) but now it can't find vcruntime140d.dll and ucrtbased.dll

And crashes.

Google tells me that 

"Those are Debug versions of DLLs.

You must compile in Release mode."

https://social.msdn.microsoft.com/Forums/windows/en-US/06b66cae-d010-441d-bc72-64a8c7cd65bc/missing-vcruntime140ddll-and-ucrtbaseddll?forum=visualstudiogeneral

So this might be the case:)

**Comment by Codectory** [comment]
I compiled it in release mode. I attached the binaries and will release it when your test was successful.
Thank very much for your help!
[Release_HDRProfile_1-3-3_x64.zip](https://github.com/Codectory/HDR-Profile/files/5824704/Release_HDRProfile_1-3-3_x64.zip)
[Release_HDRProfile_1-3-3_x86.zip](https://github.com/Codectory/HDR-Profile/files/5824705/Release_HDRProfile_1-3-3_x86.zip)

**Comment by dmgudkov** [comment]
Same exact thing as previously.

Doesn't pick up embedded HDRcontroller.dll. Tries to look for it, fails and crashes.

If i put the dll in the folder - tries to find  vcruntime140d.dll and ucrtbased.dll, fails, crashes.

There seem to be some problem with embedding those in your build.

**Comment by Codectory** [comment]
Try this: https://www.microsoft.com/en-us/download/details.aspx?id=48145

**Comment by Codectory** [comment]
I've added the missing dlls to the project.

[Release_HDRProfile_1-3-3_x64.zip](https://github.com/Codectory/HDR-Profile/files/5824752/Release_HDRProfile_1-3-3_x64.zip)
[Release_HDRProfile_1-3-3_x86.zip](https://github.com/Codectory/HDR-Profile/files/5824753/Release_HDRProfile_1-3-3_x86.zip)

**Comment by dmgudkov** [comment]
Visual C++ 15 won't install.

Error 0x80070666: Cannot install a product when a newer version is installed.

The last version doesn't start. Crash on startup.

**Comment by Codectory** [comment]
Maybe this: https://support.microsoft.com/de-de/help/2977003/the-latest-supported-visual-c-downloads

**Comment by Codectory** [comment]
Dit this solve the problem?

**Comment by Codectory** [comment]
I assume this as fixed. Please contact me again, if you have other issues.

**Comment by dmgudkov** [comment]
The libraries had no effect.

I still can't run the program.

**Comment by Codectory** [comment]
Very strange. Windows is up to date? Did you exclude the application in
your anti virus program?

dmgudkov <notifications@github.com> schrieb am Fr. 22. Jan. 2021 um 00:04:

**Comment by dmgudkov** [comment]
### Exception
1/21/2021 6:23:38 PM
	The system cannot find the file specified. (Exception from HRESULT: 0x80070002)

   at Microsoft.Win32.TaskScheduler.V2Interop.ITaskFolder.DeleteTask(String Name, Int32 flags)
   at Microsoft.Win32.TaskScheduler.TaskFolder.DeleteTask(String name, Boolean exceptionOnNotExists)
   at HDRProfile.Tools.SetAutoStartInScheduler(String applicationName, String filePath, Boolean autostart)
   at HDRProfile.HDRProfileHandler.Settings_PropertyChanged(Object sender, PropertyChangedEventArgs e)
### 

That's the current exception. I clicked "Activate HDR".

The folder is excluded in the antivirus.

Using last version. 1.4.0.

**Comment by dmgudkov** [comment]
This time it didn't crash.

I'm not sure how you approach it, but i have 3 monitors and only one of them (primary) supports hdr.

If your program fails to select the proper screen - it might explain some problems.

**Comment by dmgudkov** [comment]
Actually the exception didn't happen on clicking the button. I found it in the logs and assumed it did.

Clicking the button (or actually selecting and starting an application to trigger HDR) produces nothing in logs. It just doesn't happen.

**Comment by dmgudkov** [comment]
Ok, i took an effort to look in the HDRController.cpp code. I have a limited experience with c++ code, but it seems to be that when it both tries to set HDR mode and checks it - it doesn't go through all connected displays, nor it tries to at least make sure that it read the primary display info.

It just overwrites the info from each mode to the same variables. So after the loop the variable contains the HDR status of the last active display in the loop (or tries to toggle HDR on the last active display in the loop).

To test this i disconnected all the displays, but one (with HDR) and your program started to properly switch its HDR state (and display its status).

So it seems you need to modify the HDRController code to find the primary display and manipulate only it as easy solution.

Or you might want to go all the way and ask the user to select which display to manipulate say from a dropdown list and then still modify the HDRController.cpp to take that as an input.

TL;DR

HDRController only properly does its job if a single monitor is connected to the computer.

Needs to be modified to accommodate for people with more than one connected display.

**Comment by Codectory** [comment]
Thank you very much! As I don’t have multiple monitors, this issue won’t ever be occurred on my PC.  
Just to be sure, activating HDR in Windows works properly while only one HDR monitor is connected? What happens with the other ones? I will try to fix this after work. Thanks again for that effort!

**Comment by dmgudkov** [comment]
Precisely.

With only one monitor enabled (hdr one) your activate hdr button correctly puts it to hdr mode and back. And the status indicator correctly shows the hdr state.

With other monitors enabled - nothing happens.

In the source code it literally interates through all active display targets in the order the api function returns them and assigns their id to the loop variable.

So whatever comes last will be the monitor it *tries* to activate hdr on.

So nothing happens unless the last monitor actually supports hdr, or there’s only one monitor connected (that also have hdr).

I know you didn’t write the code for that library and since i tested the HDR switch before (where the code came from) i can assure you that it never worked for me as well:)

**Comment by Codectory** [comment]
Please try this one
[Release_HDRProfile_1-4-1_x64.zip](https://github.com/Codectory/HDR-Profile/files/5854113/Release_HDRProfile_1-4-1_x64.zip)

**Comment by dmgudkov** [comment]
First partial success!:)

It activates HDR with all screens on by both clicking "Activate HDR" button and by running an added application.

But.

1. It doesn't deactivate it. 
 
Say when a game exits. Nor by clicking the button again.

I'd suggest to make the button a toggle instead of "HDR on only" and also if possible to make a hotkey for it. So while your app is running users would be able to hotkey toggle hdr at will. This feature is greatly missed from win10 hdr interface implementation.

2. The indicator next to the "Activate HDR" doesn't change its state regardless of HDR state.

3. Could you please explain what "Start in HDR-Mode" button and "Restart on first start" checkbox are supposed to do?:)

4. Minor thing. The text on "Exit application" button is larger than the button itself and thus clips:)

Btw thank you for your work. There's literally zero solutions available to make it easier to activate/deactivate HDR in windows, besides manually going to the menu and flipping the toggle.

**Comment by Codectory** [comment]
1. The attached release should fix it.
2. see 1
3.
**Restart on first start:** In some games, e.g. Cyberpunk 2077,  you don't see any HDR options in settings, when HDR was not enabled in Windows before the game was launched. Therefore, the application kills the process on the first occurence, activates hdr and restarts the process.
**Start in HDR-Mode:** You can use HDRProfile as a launcher by selecting the application and clicking Start in HDR-Mode. This will enable HDR before starting the application, which is necessary for some applications, e.g. Cyberpunk 2077.
[Release_HDRProfile_1-4-2_x64.zip](https://github.com/Codectory/HDR-Profile/files/5854316/Release_HDRProfile_1-4-2_x64.zip)

4. see 1

**Comment by dmgudkov** [comment]
Same thing with 1.4.2.

It doesn't deactivate, no HDR indicator working.

**Comment by Codectory** [comment]
Maybe this one:
[Release_HDRProfile_1-4-2_x64.zip](https://github.com/Codectory/HDR-Profile/files/5856476/Release_HDRProfile_1-4-2_x64.zip)

Sorry for the try & error. I am not the C++ guy, too ;-)

**Comment by dmgudkov** [comment]
No worries. I figured that you'll need iterations and somebody to test them:)

Nothing changed. No hdr indicator. Doesn't deactivated.

Btw, minor issue, when you add an application it asks for display name. If you enter anything there it still uses the file name in the list.

And also in the "Add application" window the bottom buttons clip with the bottom of the window. Probably need to be moved a bit higher.

**Comment by Codectory** [comment]
Please tell me that it's working ^^

https://github.com/Codectory/HDR-Profile/releases/tag/1.5.0

**Comment by dmgudkov** [comment]
Almost there!

The button correctly switches HDR on AND off.

The indicator seems to correctly shows the HDR state.

An added app correctly starts hdr before it runs.

But when the app quits, your program either doesn't register it (still says that the app is running) or even if it does - it doesn't deactivate hdr.

So the button activates and deactivates. An app starting activates it, but app quitting doesn't. Also focused mode doesn't deactivate hdr on losing focus.

Btw, updated ui is much better. Props:)

**Comment by Codectory** [comment]
Thanks for the props!

Running/Focused does work for me. Never had problems here.
I have to extend the logging...
Can you check whether WMI is running?
https://docs.microsoft.com/en-us/windows/win32/wmisdk/starting-and-stopping-the-wmi-service

I use it to get the process start and stop event.

Maybe it's just another multi monitor related thing, as I just put an HDMI cable in the same monitor for debugging, but "both" support HDR.

**Comment by dmgudkov** [comment]
WMI is running.

I did some tests.

1. In "focused" mode it works correctly (Yay!). If the app loses focus (or quits) it correctly disables HDR and says "No app is running" in Status tab. Might want to indicate in the status message that the app is actually running, but not in focus, but then what if multiple added apps are running? Show a list?:)

2. In "running" mode it works incorrectly. Meaning that Status tab shows "APPNAME is running" regardless of if the app is closed or not. As long as it detected that the app started once - it never clears that.

Even more, if i start in "focused" (working correctly), then close my app (Status Tab shows "No app runnning") and then switch to "running" mode - the status tab will show the name of the application that is already closed (and turns on hdr).

So it sounds like your code that detects focus works correctly, but the code that detects process running is not.

The actual HDR turning on and off seems to work correctly when the main code calls on it.

3. There's another understandable issue with applications. If the app is started using one .exe, but that .exe actually invokes another .exe, then for obvious reasons you program doesn't handle it correctly. Maybe some time in the future you might want to take a look if it's possible to trace the final .exe.

4.  There's still this error in the logs, whatever it means:)

### Exception
1/22/2021 7:39:40 PM
	The system cannot find the file specified. (Exception from HRESULT: 0x80070002)

   at Microsoft.Win32.TaskScheduler.V2Interop.ITaskFolder.DeleteTask(String Name, Int32 flags)
   at Microsoft.Win32.TaskScheduler.TaskFolder.DeleteTask(String name, Boolean exceptionOnNotExists)
   at HDRProfile.Tools.SetAutoStartInScheduler(String applicationName, String filePath, Boolean autostart)
   at HDRProfile.HDRProfileHandler.Settings_PropertyChanged(Object sender, PropertyChangedEventArgs e)
### 

PS. I checked your changes to the c++ code:) Actually it's a good fast idea to turn on hdr on all the screens. There might be some edge cases when it will break, but for now seems robust enough (unless you have multiple hdr screens and your "Check if HDR running" will only rely on positive flag in the first one in the list).

**Comment by Codectory** [comment]
1. I thought about that. Showing a list is something I will do in the future.  The idea of the Status window was, to let something happen, if the tool changes something. Quick and dirty feedback.
The focus indicator is a bit difficult, because when you want to see the indicator, the application ins't in focus anymore, but I will implement some history log in the status field, than you can  check whether the app was in focus.

2. I will check if using the .Net implementation is more practicable.
3.. Understand. Can you give me an explicit example? Using the "last" exe is not practicable?
4. I will fix this in the next update. I disable the Autostart feature with deleting the scheduled task. When there was no task, you get that exception.

C++:
The only way a can clean this up, is checking which monitor supports HDR. But that's C++ stuff, so I will may do that a bit later ;-)

Thank you very much for all that feedback!

**Comment by dmgudkov** [comment]
2. Please do. I tried to check your C# code and wasn't able to find obvious problems (even though i'm not an expert in C# as well), so it might have something to do with the actual stop events.

3. Examples might be most unity based based games. For instance i tested with Bloodstained Ritual of the Night.

It has BloodstainedRotN.exe in the game root folder that actually launches the main exe located in BloodstainedRotN\Binaries\Win64 subfolder ("BloodstainedRotN-Win64-Shipping.exe").

So the user needs to know that it is the case to add the proper exe in your app.

If the user is advanced enough he can check through task manager which exe is running and add this exe to your program.

But if the user is not computer savvy - he might be confused.

Also a typical example would be most games with launchers.

Again this is not a crucial problem. If proper exe is pinpointed - the program will work.

It's a user friendliness issue, that might be too costly to address.

**Comment by Codectory** [comment]
Are you sure that the app is really closed? A checked the WMI thing.
Since I didn't found a solution to get a start and stop event, I have to stay with it, otherwise the "restart" feature cannot work, when you start the tool after the application which needs to be restarted.

**Comment by dmgudkov** [comment]
Yes. I tried multiple apps.

The behavior in "running" mode stays.

If you can't find a culprit, can you send me a version which outputs to logs "running", "focused" and settings boolean flag in your _applications dictionary?

**Comment by dmgudkov** [comment]
and actual start/stop events for registered applications

**Comment by Codectory** [comment]
In the works!

**Comment by Codectory** [comment]
Check the new release!

**Comment by dmgudkov** [comment]
Same thing. In "running" mode on app exit, HDR profile doesn't register it.

**Comment by Codectory** [comment]
Ok, can you send me the log file?

**Comment by Codectory** [comment]
And pleas entry some other app, e.g. notepad.exe

**Comment by dmgudkov** [comment]
1/24/2021 10:24:04 AM:Initializing...
1/24/2021 10:24:04 AM:Loading settings...
1/24/2021 10:24:04 AM:Settings loaded
1/24/2021 10:24:04 AM:Initialized
1/24/2021 10:24:04 AM:Starting HDR Monitoring...
1/24/2021 10:24:04 AM:HDR Monitoring started
1/24/2021 10:24:04 AM:Starting process watcher...
1/24/2021 10:24:04 AM:Process watcher started
1/24/2021 10:24:25 AM:Application added: BloodstainedRotN-Win64-Shipping
1/24/2021 10:24:25 AM:Saving settings..
1/24/2021 10:24:25 AM:Settings saved
1/24/2021 10:24:34 AM:Start application BloodstainedRotN-Win64-Shipping
1/24/2021 10:25:04 AM:Stopping HDR Monitoring...
1/24/2021 10:25:04 AM:HDR Monitoring stopped
1/24/2021 10:25:04 AM:Stopping process watcher...
1/24/2021 10:25:04 AM:Process watcher stopped
1/24/2021 10:25:04 AM:Shutting down...

**Comment by dmgudkov** [comment]
1. Started HDR profile.
2. Added BloodstainedRotN-Win64-Shipping.exe
3. Started the game from HDR profile.
4. Exited the game after it loaded.
5. Exited HDR profile.

**Comment by dmgudkov** [comment]
1/24/2021 10:27:15 AM:Initializing...
1/24/2021 10:27:15 AM:Loading settings...
1/24/2021 10:27:15 AM:Settings loaded
1/24/2021 10:27:15 AM:Initialized
1/24/2021 10:27:15 AM:Starting HDR Monitoring...
1/24/2021 10:27:15 AM:HDR Monitoring started
1/24/2021 10:27:15 AM:Starting process watcher...
1/24/2021 10:27:15 AM:Process watcher started
1/24/2021 10:27:19 AM:Start application BloodstainedRotN-Win64-Shipping

Without step 5.

So when the app exits, your logs have nothing.

**Comment by Codectory** [comment]
Can you start the game as usual? Not through HDR profile?

**Comment by dmgudkov** [comment]
1/24/2021 10:39:56 AM:Initializing...
1/24/2021 10:39:56 AM:Loading settings...
1/24/2021 10:39:56 AM:Settings loaded
1/24/2021 10:39:56 AM:Initialized
1/24/2021 10:39:56 AM:Starting HDR Monitoring...
1/24/2021 10:39:56 AM:HDR Monitoring started
1/24/2021 10:39:56 AM:Starting process watcher...
1/24/2021 10:39:56 AM:Process watcher started
1/24/2021 10:40:02 AM:Activating HDR...
1/24/2021 10:40:03 AM:HDR is active

**Comment by dmgudkov** [comment]
That's the "usual" for steps 1-4.

**Comment by Codectory** [comment]
So you started bloodstained from Windows not hdr profile?

**Comment by dmgudkov** [comment]
Yes, as you asked.

**Comment by dmgudkov** [comment]
Yes, there're no entries in your logs, when the app exits.

That's why i asked for a version with more detailed logs.

**Comment by Codectory** [comment]
Understand, I implanted extended logging in 1.5.2. Could you test if the behaviour is the same with notepad.exe for example?

**Comment by dmgudkov** [comment]
Good point.

With notepad .exe it properly activated AND deactivates hdr.

Let me test more games to see if i can find a pattern.

**Comment by dmgudkov** [comment]
Ok.

It's all over the place.

Most games indeed correctly deactivate hdr on exit.

I happen to use 2 of those to test since they are lightweight and start fast. And both of those were those that did NOT deactivate HDR.

So far deactivation doesn't work for:

Baba is you.
Bloodstrained.
Hellblade sacrifice.

**Comment by dmgudkov** [comment]
and Sinking City

**Comment by dmgudkov** [comment]
So it seems, that some games doesn't properly send the stop event on exit.

You might want to find an alternative way to track that.

**Comment by Codectory** [comment]
New day, new luck. I uploaded 1.5.3, which is using .Net process monitoring (not event triggered).

**Comment by dmgudkov** [comment]
I think i have to congratulate you:)

Everything works perfectly so far.

Including exe running a second exe.

On app exit hdr properly deactivates.

Thank you for putting efforts into this:)

I did a quick test, but now i'll try actually use it and will report any problems.

**Comment by Codectory** [comment]
Nice to hear that! Thank you four your help! I wouldn‘t have found these issues and fox them so quick. 
I will close this, as “multi monitor” seems to work now.
